### ISSUE TITLE
Dumb jump slow lookup if not used for a while

### ISSUE THREAD

**Issue by nerton** [jacktasia/dumb-jump#282]
Dumb jump usually works nicely, but I noticed it is very slow to look up things if not used for a while. It takes several seconds for it to get a result then it's quick again.

I'm trying it on Windows with ripgrep, so at first I thought it is because the source files were ejected from the memory cache  and Windows is slow to read the files again from disk.

But then I turned on dumb jump debugging, copied the ripgrep search command  line and tried it in a command prompt. It worked quickly.

Then I did something else for an hour, not using dumb jump at all, then tried again in the console with the same search command and the results were instantaneous again.

Then I switched to emacs and tried dumb jump from there and it was very slow again the first time, took several seconds.

I don't really get the difference. If it's the windows file cache then it should have been slow from the command prompt too after not using it for a while. But it was quick there, so it must be something what dumb jump does.

Do you do some kind of internal caching or computation which is thrown away after a while and has to be recreated?

**Comment by jacktasia** [comment]
> Do you do some kind of internal caching or computation which is thrown away after a while and has to be recreated?

No, at least not on the dumb-jump side. It's possible the searcher you're using does, but that would be news to me. For the most part speed should be a function of the size of project it's searching. Was all this testing on the same project?

**Comment by nerton** [comment]
Yes, same project. I copied the command line printed by dumb jump's debug, so it was the same project, same command line, same search tool (ripgrep), everything.

The only difference is that it was invoked via emacs with dumb jump. I'l try to profile the dumb jump call, to see which part takes seconds.

**Comment by jacktasia** [comment]
> The only difference is that it was invoked via emacs with dumb jump. I'l try to profile the dumb jump call, to see which part takes seconds.

Gotcha, I suppose it's possible that in some cases the response from  ipgrep is so large it's taking a non-trivial time to parse and/or it's getting blocked by something else in your emacs since it's single-threaded.

**Comment by nerton** [comment]
The response from ripgrep is a single line. I see you use  shell-command-to-string . I copied the same search command line which dumb-jump-debug prints and invoked it with shell-command-to-string directly and it took 3-4 seconds to return that a single line. The same command started from a windows command prompt returns instantly.

So it looks like shell-command-to-string is the problem. I heard in the past that emacs i/o can be slow on windows, but I never noticed this much of a difference. Anyway, it looks like dumb jump is not at fault here, since using the relevant emacs function directly also demonstrates the same slowness.

**Comment by jacktasia** [comment]
> Anyway, it looks like dumb jump is not at fault here, since using the relevant emacs function directly also demonstrates the same slowness.

Thanks for looking into this! FWIW, I feel like I've never had this problem on linux. I'll close this issue for now, but please re-open it or create another issue if you see anything else strange. Thanks!

**Comment by nerton** [comment]
Just a heads up if somebody comes around with the same problem later. It was becasuse of WIndows Defender. That slowed down ripgrep when called via Emacs. Disabling Windows Defender for the project directory or for the ripgrep process makes the delay go away.

**Comment by jacktasia** [comment]
@nerton Wow, very interesting. Thanks so much for coming back and closing the loop on this.
